### PR TITLE
Fix checks for `needs` in Actions

### PR DESCRIPTION
**Pull request by wolfogre**
Fix:
- https://gitea.com/gitea/act_runner/issues/77
- https://gitea.com/gitea/act_runner/issues/81

Before:
image

Highlights:
- Upgrade act to make things doable, related to
  - https://gitea.com/gitea/act/pulls/32
  - https://gitea.com/gitea/act/pulls/33
  - https://gitea.com/gitea/act/pulls/35
- Make `needs` works
- Sort jobs in the original order in the workflow files

### CODE REVIEW

**Review by lunny — APPROVED** [review]

File: web_src/js/components/RepoActionView.vue [hunk 1]
```diff
@@ -405,6 +405,7 @@ export function initRepositoryActionView() {
 
 .job-group-section .job-brief-list .job-brief-item .job-brief-link span {
   margin-right: 8px;
+  padding-left: 5px;
```
Comment (silverwind) [inline web_src/js/components/RepoActionView.vue]:
```suggestion
  margin: 0 8px;
```

Equal margin on both sides, and I do think 5px is too few. BTW, `gt-mx-3` helper would achieve the same.

**Comment by silverwind** [comment]
BTW, Stop button is vertically misaligned:

image

If this is a flexbox, use `align-content: center`, otherwise, play with `vertical-align`.

**Comment by yp05327** [comment]
It seems that https://github.com/go-gitea/gitea/issues/23680#issuecomment-1482358330 will be fixed in this PR?

**Review by silverwind — APPROVED** [review]

**Comment by wolfogre** [comment]
It seems that @GiteaBot is offline. 🤔

**Pull request merged**